src/main/java/org/akaza/openclinica/controller/SdvApiController.java

```diff
@@ -62,7 +69,18 @@ else if(ordinal.matches("^[0-9]*$") && Integer.parseInt(ordinal) > 0)
                 ordinalValue = Integer.parseInt(ordinal);
             else
                 throw new OpenClinicaSystemException( ErrorConstants.ERR_EVENT_ORDINAL_IS_INCORRECT);
-            validateService.validateForSdvItemForm(studyOID, studyEventOID, studySubjectLabel, formOID, userAccountBean,ordinalValue);
+
+            EventCrf eventCrf = eventCrfDao.findByStudyEventOIdStudySubjectOIdCrfOId(studyEventOID, studySubjectLabel, formOID, ordinalValue);
+            boolean hasAccess = permissionService.hasFormAccess(
+                eventCrf, 
+                eventCrf.getFormLayout().getFormLayoutId(), 
+                eventCrf.getStudyEvent().getStudyEventId(), 
+                request
+            );
+            if (!hasAccess)
+                throw new OpenClinicaSystemException(ErrorConstants.ERR_HAS_NO_ACCESS_TO_FORM);
+                
```

Comment:
this permission check logic should go into the validateForSdvItemForm. It is usually not a good idea to add business logic to Controller classes.

Reply:
OK permission check moved to validateForSdvItemForm()